flocker/control/_model.py

```diff
@@ -625,6 +625,47 @@ def update_cluster_state(cluster_state):
             with changes from this object applied to it.
         """
 
+    def get_information_wipe():
+        """
+        Create a ``IClusterStateWipe`` that can wipe information added by
+        this change.
+
+        For example, if this update adds information to a particular node,
+        the returned ``IClusterStateChange`` will wipe out that
+        information indicating ignorance about that information. We need
+        this ability in order to expire out-of-date state information.
+
+        :return: A ``IClusterStateWipe`` that undoes this update.
+        """
+
+
+class IClusterStateWipe(Interface):
+    """
+    An ``IClusterStateWipe`` can remove some information from a
+    ``DeploymentState``.
+    """
+    def update_cluster_state(cluster_state):
+        """
+        :param DeploymentState cluster_state: Some current known state of the
+            cluster.
+
+        :return: A new ``DeploymentState`` similar to ``cluster_state`` but
+            with some information removed from it.
+        """
+
+    def key():
+        """
+        Return a key describing what information will be wiped.
+
+        Providers that wipe the same information should return the same
+        key, and providers that wipe different information should return
+        differing keys.
+
+        Different ``IClusterStateWipe`` implementors are presumed to
+        cover different information, so there is no need for the key to
+        express that differentation.
```

Comment:
Implied here is that the type of the object that provides the interface is part of the interface.  Probably worth stating this explicitly.